Mono.Debugging/Mono.Debugging.Client/DebuggerStatistics.cs

```diff
@@ -120,10 +126,18 @@ public class DebuggerTimer : IDisposable
 		readonly DebuggerStatistics stats;
 		readonly Stopwatch stopwatch;
 
-		public DebuggerTimer (DebuggerStatistics stats)
+		static readonly TraceListener traceListener = new TraceSource (nameof (DebuggerTimer)).Listeners.OfType<TraceListener> ().FirstOrDefault (l => l.Name == nameof (DebuggerTimer));
+		static int traceSeq = 0;
+		int traceId;
+
+		public DebuggerTimer (DebuggerStatistics stats, string name)
 		{
-			stopwatch = new Stopwatch ();
+			stopwatch = Stopwatch.StartNew ();
 			this.stats = stats;
+			if (stats != null && traceListener != null) {
+				traceId = Interlocked.Increment (ref traceSeq);
+				traceListener?.TraceEvent (null, stats.Name, TraceEventType.Start, traceId, name);
```

Comment:
Already null checked.

```suggestion
				traceListener.TraceEvent (null, stats.Name, TraceEventType.Start, traceId, name);
```